### PR TITLE
feat(delegate): Add create_sub_visualizer method for custom sub-agent visualization

### PR DESCRIPTION
**Pull request by malhotra5**
## Summary

This PR adds a `create_sub_visualizer()` method to the visualizer base class that allows custom visualizers (like TUI-based ones) to create appropriate sub-visualizers for delegate sub-agents.

## Problem

The current implementation in `DelegateExecutor._spawn_agents()` uses an `isinstance()` check against `DelegationVisualizer` to determine if sub-agent visualization should be enabled:

```python
if isinstance(parent_visualizer, DelegationVisualizer):
    sub_visualizer = DelegationVisualizer(...)
else:
    sub_visualizer = None
```

This approach prevents custom visualizers (like the CLI's Textual-based visualizer) from providing their own sub-agent visualization implementation, since they don't inherit from `DelegationVisualizer`.

## Solution

Add a `create_sub_visualizer(agent_id: str)` method to the visualizer base class:

1. **`ConversationVisualizerBase.create_sub_visualizer()`** - Returns `None` by default (no sub-agent visualization)
2. **`DelegationVisualizer.create_sub_visualizer()`** - Overrides to create a new `DelegationVisualizer` for the sub-agent
3. **`DelegateExecutor._spawn_agents()`** - Updated to call `parent_visualizer.create_sub_visualizer(agent_id)` instead of using `isinstance()` checks

This allows any custom visualizer to implement sub-agent visualization by overriding the `create_sub_visualizer()` method.

## Changes

- `openhands-sdk/openhands/sdk/conversation/visualizer/base.py`: Add `create_sub_visualizer()` method returning `None` by default
- `openhands-tools/openhands/tools/delegate/visualizer.py`: Override `create_sub_visualizer()` to create sub-agent visualizers
- `openhands-tools/openhands/tools/delegate/impl.py`: Use `create_sub_visualizer()` method instead of `isinstance()` checks
- `tests/sdk/conversation/test_visualizer.py`: Add test for base class returning `None`
- `tests/tools/delegate/test_visualizer.py`: Add tests for `DelegationVisualizer.create_sub_visualizer()`

## Testing

All existing tests pass, plus new tests added:
- `test_default_visualizer_create_sub_visualizer_returns_none`
- `test_delegation_visualizer_create_sub_visualizer`
- `test_delegation_visualizer_create_sub_visualizer_with_defaults`

## Usage Example

For a custom TUI visualizer in the CLI:

```python
class CLIDelegationVisualizer(ConversationVisualizer):
    def __init__(self, container, app, name=None, ...):
        super().__init__(container, app, ...)
        self._name = name
        
    def create_sub_visualizer(self, agent_id: str) -> "CLIDelegationVisualizer":
        return CLIDelegationVisualizer(
            container=self._container,  # Share the same TUI container
            app=self._app,
            name=agent_id,
            # ... other settings
        )
```

@malhotra5 can click here to [continue refining the PR](https://app.all-hands.dev/conversations/None)

---
**Agent Server images for this PR**

• **GHCR package:** https://github.com/OpenHands/agent-sdk/pkgs/container/agent-server

**Variants & Base Images**
| Variant | Architectures | Base Image | Docs / Tags |
|---|---|---|---|
| java | amd64, arm64 | `eclipse-temurin:17-jdk` | [Link](https://hub.docker.com/_/eclipse-temurin:17-jdk) |
| python | amd64, arm64 | `nikolaik/python-nodejs:python3.12-nodejs22` | [Link](https://hub.docker.com/_/nikolaik/python-nodejs:python3.12-nodejs22) |
| golang | amd64, arm64 | `golang:1.21-bookworm` | [Link](https://hub.docker.com/_/golang:1.21-bookworm) |

**Pull (multi-arch manifest)**
```bash
# Each variant is a multi-arch manifest supporting both amd64 and arm64
docker pull ghcr.io/openhands/agent-server:ee2b55a-python
```

**Run**
```bash
docker run -it --rm \
  -p 8000:8000 \
  --name agent-server-ee2b55a-python \
  ghcr.io/openhands/agent-server:ee2b55a-python
```

**All tags pushed for this build**
```
ghcr.io/openhands/agent-server:ee2b55a-golang-amd64
ghcr.io/openhands/agent-server:ee2b55a-golang_tag_1.21-bookworm-amd64
ghcr.io/openhands/agent-server:ee2b55a-golang-arm64
ghcr.io/openhands/agent-server:ee2b55a-golang_tag_1.21-bookworm-arm64
ghcr.io/openhands/agent-server:ee2b55a-java-amd64
ghcr.io/openhands/agent-server:ee2b55a-eclipse-temurin_tag_17-jdk-amd64
ghcr.io/openhands/agent-server:ee2b55a-java-arm64
ghcr.io/openhands/agent-server:ee2b55a-eclipse-temurin_tag_17-jdk-arm64
ghcr.io/openhands/agent-server:ee2b55a-python-amd64
ghcr.io/openhands/agent-server:ee2b55a-nikolaik_s_python-nodejs_tag_python3.12-nodejs22-amd64
ghcr.io/openhands/agent-server:ee2b55a-python-arm64
ghcr.io/openhands/agent-server:ee2b55a-nikolaik_s_python-nodejs_tag_python3.12-nodejs22-arm64
ghcr.io/openhands/agent-server:ee2b55a-golang
ghcr.io/openhands/agent-server:ee2b55a-java
ghcr.io/openhands/agent-server:ee2b55a-python
```

**About Multi-Architecture Support**
- Each variant tag (e.g., `ee2b55a-python`) is a **multi-arch manifest** supporting both **amd64** and **arm64**
- Docker automatically pulls the correct architecture for your platform
- Individual architecture tags (e.g., `ee2b55a-python-amd64`) are also available if needed

### CODE REVIEW

**Review by xingyaoww — APPROVED** [review]
LGTM!

File: openhands-tools/openhands/tools/delegate/impl.py [hunk 1]
```diff
@@ -211,13 +208,13 @@ def _delegate_tasks(self, action: "DelegateAction") -> "DelegateObservation":
...
-            # Get the parent agent's name from the visualizer
+            # Get the parent agent's name from the visualizer if available
             parent_conversation = self.parent_conversation
             parent_name = None
             if hasattr(parent_conversation, "_visualizer"):
                 visualizer = parent_conversation._visualizer
-                if isinstance(visualizer, DelegationVisualizer):
-                    parent_name = visualizer._name
+                if visualizer is not None:
+                    parent_name = getattr(visualizer, "_name", None)
```
[the remark this answers is not in the document]
Comment (xingyaoww) [reply openhands-tools/openhands/tools/delegate/impl.py:217]: ^^ optional NIT: ideally we should fix this. We can make `.name` a property.
